things/clients/redis/events.go

```diff
@@ -17,75 +29,289 @@ var (
 	_ event = (*createClientEvent)(nil)
 	_ event = (*updateClientEvent)(nil)
 	_ event = (*removeClientEvent)(nil)
+	_ event = (*shareClientEvent)(nil)
+	_ event = (*viewClientEvent)(nil)
+	_ event = (*listClientEvent)(nil)
+	_ event = (*listClientByGroupEvent)(nil)
+	_ event = (*identifyClientEvent)(nil)
 )
 
 type createClientEvent struct {
-	id       string
-	owner    string
-	name     string
-	metadata map[string]interface{}
+	mfclients.Client
 }
 
-func (cte createClientEvent) Encode() map[string]interface{} {
+func (cce createClientEvent) Encode() map[string]interface{} {
 	val := map[string]interface{}{
-		"id":        cte.id,
-		"owner":     cte.owner,
-		"operation": thingCreate,
+		"operation":  clientCreate,
+		"id":         cce.ID,
+		"status":     cce.Status.String(),
+		"created_at": cce.CreatedAt,
 	}
 
-	if cte.name != "" {
-		val["name"] = cte.name
+	if cce.Name != "" {
+		val["name"] = cce.Name
 	}
-
-	if cte.metadata != nil {
-		metadata, err := json.Marshal(cte.metadata)
+	if len(cce.Tags) > 0 {
+		tags := fmt.Sprintf("[%s]", strings.Join(cce.Tags, ","))
+		val["tags"] = tags
+	}
+	if cce.Owner != "" {
+		val["owner"] = cce.Owner
+	}
+	if cce.Metadata != nil {
+		metadata, err := json.Marshal(cce.Metadata)
 		if err != nil {
 			return val
 		}
 
-		val["metadata"] = string(metadata)
+		val["metadata"] = metadata
+	}
+	if cce.Credentials.Identity != "" {
+		val["identity"] = cce.Credentials.Identity
 	}
 
 	return val
 }
 
 type updateClientEvent struct {
-	id       string
-	name     string
-	owner    string
-	tags     []string
-	metadata map[string]interface{}
+	mfclients.Client
+	operation string
 }
 
-func (ute updateClientEvent) Encode() map[string]interface{} {
+func (uce updateClientEvent) Encode() map[string]interface{} {
 	val := map[string]interface{}{
-		"id":        ute.id,
-		"operation": thingUpdate,
+		"operation":  clientUpdate + "." + uce.operation,
+		"updated_at": uce.UpdatedAt,
+		"updated_by": uce.UpdatedBy,
 	}
 
-	if ute.name != "" {
-		val["name"] = ute.name
+	if uce.ID != "" {
+		val["id"] = uce.ID
 	}
-
-	if ute.metadata != nil {
-		metadata, err := json.Marshal(ute.metadata)
+	if uce.Name != "" {
+		val["name"] = uce.Name
+	}
+	if len(uce.Tags) > 0 {
+		tags := fmt.Sprintf("[%s]", strings.Join(uce.Tags, ","))
+		val["tags"] = tags
+	}
+	if uce.Credentials.Identity != "" {
+		val["identity"] = uce.Credentials.Identity
+	}
+	if uce.Metadata != nil {
+		metadata, err := json.Marshal(uce.Metadata)
```

Comment:
We're masking the error here.

Reply:
Yes

Reply:
Please don't mask the error. :)

Reply:
Should `Encode()` return `(error, map[string]interface{})` ?